### PR TITLE
Create credit card filter method

### PR DESCRIPTION
**Pull request by Seralto**
## Task
Filter credit card numbers in the WordPress plugin conversions.

### CODE REVIEW

File: includes/resources/rdsm_conversion.php [hunk 1]
```diff
@@ -63,13 +63,42 @@ public function build_payload($form_data) {
...
+    if (preg_match($cardtype['amex'], $number) ||
+        preg_match($cardtype['diners'], $number) ||
+        preg_match($cardtype['discover'], $number) ||
+        preg_match($cardtype['elo'], $number) ||
+        preg_match($cardtype['mastercard'], $number) ||
+        preg_match($cardtype['visa'], $number)) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (filipenasc) [inline includes/resources/rdsm_conversion.php:99]: What do you think about looping all the card types to validate them?

This way, when we add a new card type, we'll only need to change the `$cardtypes` array.
```suggestion
foreach($cardtypes as $cardtype) {
    $credit_card_number = preg_match($cardtype, $number);

    if ($credit_card_number) {
      return true;
    }
  }
  
  return false;
```

File: includes/resources/rdsm_conversion.php [hunk 2]
```diff
@@ -63,13 +63,42 @@ public function build_payload($form_data) {
...
       }
     }
+
     return $form_fields;
   }
 
+  private function is_credit_card_number($number)  {
+    $number = preg_replace('/\D/', '', $number);
+
+    $cardtype = array(
```
Comment (filipenasc) [inline includes/resources/rdsm_conversion.php:81]:
```suggestion
    $cardtypes = array(
```

**Review by filipenasc — COMMENTED** [review]
Do you think that now we can remove these lines safely?

https://github.com/ResultadosDigitais/rdstation-wp/blob/master/includes/resources/rdsm_conversion.php#L22-L36

File: includes/resources/rdsm_conversion.php [hunk 3]
```diff
@@ -63,13 +63,41 @@ public function build_payload($form_data) {
...
+    $cardtypes = array(
+      "amex"       => "/^3[47][0-9]{13}$/",
+      "diners"     => "/^3(?:0[0-5]|[68][0-9])[0-9]{11}$/",
+      "discover"   => "/^6(?:011|5[0-9]{2})[0-9]{12}$/",
+      "elo"        => "/^((((636368)|(438935)|(504175)|(451416)|(636297)(506699)|(636369))\d{0,10})|((5067)|(4576)|(4011))\d{0,12})/",
+      "mastercard" => "/^5[1-5][0-9]{14}$/",
+      "visa"       => "/^(5[1-5]|222[1-9]|22[3-9]|2[3-6]|27[01]|2720)[0-9]{14}$/"
+    );
+
+foreach($cardtypes as $cardtype) {
```
Comment (Seralto) [inline includes/resources/rdsm_conversion.php:90]:
```suggestion
  foreach($cardtypes as $cardtype) {
```

File: includes/resources/rdsm_conversion.php [hunk 4]
```diff
@@ -63,13 +63,41 @@ public function build_payload($form_data) {
...
+      "visa"       => "/^(5[1-5]|222[1-9]|22[3-9]|2[3-6]|27[01]|2720)[0-9]{14}$/"
+    );
+
+  foreach($cardtypes as $cardtype) {
+    $credit_card_number = preg_match($cardtype, $number);
+
+    if ($credit_card_number) {
+      return true;
+    }
+  }
```
Comment (Seralto) [inline includes/resources/rdsm_conversion.php:96]:
```suggestion
    foreach($cardtypes as $cardtype) {
      $credit_card_number = preg_match($cardtype, $number);

      if ($credit_card_number) {
        return true;
      }
    }
```

File: includes/resources/rdsm_conversion.php [hunk 5]
```diff
@@ -63,13 +63,41 @@ public function build_payload($form_data) {
...
+
+    foreach($cardtypes as $cardtype) {
+      $credit_card_number = preg_match($cardtype, $number);
+
+      if ($credit_card_number) {
+        return true;
+      }
+    }
+  
+  return false;
```
Comment (Seralto) [inline includes/resources/rdsm_conversion.php:98]:
```suggestion
    return false;
```

**Comment by Seralto** [comment]
I don't think so. This filter aims to remove any credit card number that came with a field name that we did not expected.
So, both are necessary.

**Review by filipenasc — APPROVED** [review]

File: includes/resources/rdsm_conversion.php [hunk 6]
```diff
@@ -63,13 +63,41 @@ public function build_payload($form_data) {
...
+      "amex"       => "/^3[47][0-9]{13}$/",
+      "diners"     => "/^3(?:0[0-5]|[68][0-9])[0-9]{11}$/",
+      "discover"   => "/^6(?:011|5[0-9]{2})[0-9]{12}$/",
+      "elo"        => "/^((((636368)|(438935)|(504175)|(451416)|(636297)(506699)|(636369))\d{0,10})|((5067)|(4576)|(4011))\d{0,12})/",
+      "mastercard" => "/^5[1-5][0-9]{14}$/",
+      "visa"       => "/^(5[1-5]|222[1-9]|22[3-9]|2[3-6]|27[01]|2720)[0-9]{14}$/"
+    );
+
+    foreach($cardtypes as $cardtype) {
+      $credit_card_number = preg_match($cardtype, $number);
```
Comment (filipenasc) [inline includes/resources/rdsm_conversion.php:91]: I've received this error when testing on a production blog:

`preg_match() expects parameter 2 to be string, array given`

**Review by filipenasc — CHANGES_REQUESTED** [review]

**Review by filipenasc — APPROVED** [review]

**Review by rafaelhancke — COMMENTED** [review]
LGTM

**Pull request merged**